include/deal.II/lac/vector_space_vector.h

```diff
@@ -0,0 +1,181 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__vector_space_vector_h
+#define dealii__vector_space_vector_h
+
+DEAL_II_NAMESPACE_OPEN
+
+#include <deal.II/base/mpi.h>
+#include <deal.II/base/numbers.h>
+
+namespace LinearAlgebra
+{
+  template <typename Number>
+  class ReadWriteVector<Number>;
+
+  class IndexSet;
+
+  /*! @addtogroup Vectors
+   *@{
+   */
+
+  /**
+   * VectorSpaceVector is an abstract class that is used to define the interface
+   * that vector classes need to implement when they want to implement global
+   * operations. This class is complementary of ReadWriteVector which allows
+   * the access of individual elements but does not allow global operations.
+   *
+   * @author Bruno Turcksin, 2015.
+   */
+  template <Number>
+  class VectorSpaceVector
+  {
+  public:
+    typedef types::global_dof_index                           size_type;
+    typedef typename numbers::NumberTraits<Number>::real_type real_type;
```

Comment:
There should probably also be a typedef to value_type in here; I guess there is no need for the full set of iterators/pointers because those are implementation-specific and do not make sense in a base class.